### ISSUE TITLE
Error calling stripe fetch Invariant Violation: hostEndpoint has not been initialized

### ISSUE THREAD

**Issue by vrugtehagel** [stripe/stripe-apps#373]
**Describe the bug**
I've been trying to use the secret store for accounts, but it throws the following error:
```
api_fetch.ts:38 error calling stripe fetch Invariant Violation: hostEndpoint has not been initialized
    at invariant (http://localhost:4242/.build/ext.js:6063:22)
    at getHostEndpoint (http://localhost:4242/.build/ext.js:6088:33)
    at StripeTailorHttpClient.stripeApiFetch [as _fetch] (http://localhost:4242/.build/ext.js:6103:50)
    at StripeTailorHttpClient.<anonymous> (http://localhost:4242/.build/ext.js:6193:37)
    at Generator.next (<anonymous>)
    at http://localhost:4242/.build/ext.js:6141:73
    at new Promise (<anonymous>)
    at __awaiter (http://localhost:4242/.build/ext.js:6123:16)
    at StripeTailorHttpClient.makeRequest (http://localhost:4242/.build/ext.js:6180:18)
    at makeRequest (http://localhost:4242/.build/ext.js:2849:64)
```
This error seems to only occur when immediately trying to fetch a secret - if I wait a second after the extension has loaded, the secret store functions properly without the above error.

**To Reproduce**
Copy the example code from the [documentation](https://stripe.com/docs/stripe-apps/store-auth-data-custom-objects) into a project. Change the `scope[type]` to `account` and remove the `scope[user]`.

For me, I created a new file containing
```ts
import Stripe from 'stripe';
import {createHttpClient, STRIPE_API_KEY} from '@stripe/ui-extension-sdk/http_client';

(async () => {
    function secretPathHelper(secretName: string): string {
      return `apps/secrets?scope[type]=account&name=${secretName}`;
    }

    // Create an instance of a Stripe object to access customer information.
    // You don't need an API Key here, Stripe supplies it for you.
    const stripe: Stripe = new Stripe(STRIPE_API_KEY, {
      httpClient: createHttpClient() as Stripe.HttpClient,
      apiVersion: '2020-08-27',
    });

    // Create the Secret Store API URL path.
    const apiPath: string = secretPathHelper('secret_name');

    // Create a `StripeResource` to load the custom Secret Store API endpoint.
    const createSecretResource = Stripe.StripeResource.extend({
        request: Stripe.StripeResource.method({
            method: 'POST',
            path: apiPath,
        })
    });
    const secretResource: Stripe.StripeResource = new createSecretResource(stripe);

    // The secret value that you store in the Secret Store API.
    const secretValue: string = 'your secret here';

    // Add secret to the Secret Store API; returns either error or secret object.
    try {
      await secretResource.request({payload: secretValue});
      console.log('hurray!')
    } catch (e) {
      // If there’s an error while setting the secret, handle it here.
      console.log(e)
    }
})()
```
Basically, this is just the example code wrapped in an async IIFE, changed the scope to `account`, and added some console logs at the end. This however throws with the aforementioned error.

**Expected behavior**
I expect the code to function without (unintended) errors.

**Desktop (please complete the following information):**
 - OS: MacOS Monterey
 - Browser: Chrome
 - Version: 99.0.4844.84 (Official Build) (x86_64)

**Additional context**
If it is not possible for this to work as-is, perhaps there should be a way to wait for the hostEndpoint to be initialized through e.g. a promise or event listener (if there isn't one already).
